libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -176,6 +176,20 @@ void ExpressionCompiler::appendStateVariableAccessor(VariableDeclaration const&
 	m_context.appendJump(eth::AssemblyItem::JumpType::OutOfFunction);
 }
 
+bool ExpressionCompiler::visit(Conditional const& _condition)
+{
+	CompilerContext::LocationSetter locationSetter(m_context, _condition);
+	_condition.condition().accept(*this);
+	eth::AssemblyItem trueTag = m_context.appendConditionalJump();
+	_condition.falseExpression().accept(*this);
```

Comment:
Oops, I was totally unaware of the necessity of conversion here. Thanks for reminding, I'll add a test for this.